api/app/controllers/transaction.controller.js

```diff
@@ -0,0 +1,137 @@
+const db = require("../sequelize/models");
+const { TRANSACTION_STATUS, TRANSACTION_TYPE } = require("../utils/constants");
+const { generateTransactionId } = require("../utils/transaction");
+
+const Transaction = db.transaction;
+const User = db.user;
+const Review = db.review;
+const Wallet = db.wallet;
+
+const Op = db.Sequelize.Op;
+
+// Create and Save a new Transaction
+exports.create = async (req, res) => {
+  // check if any fields are empty
+  const requiredFields = ["userId", "amount", "transactionType"];
+  if (
+    req.body.transactionType &&
+    req.body.transactionType === TRANSACTION_TYPE.CREDIT
+  ) {
+    requiredFields.push("reviewId");
+  }
+  const missingFields = requiredFields.filter((field) => !req.body[field]);
+  if (missingFields.length) {
+    res.status(400).send({
+      message: `Missing required fields: ${missingFields.join(", ")}`,
+    });
+    return;
+  }
+
+  const currentTime = new Date();
+
+  // check if review exists
+  if (req.body.reviewId) {
+    const review = await Review.findOne({
+      where: {
+        id: req.body.reviewId,
+      },
+    });
+    if (!review) {
+      res.status(404).send({
+        message: `Could not create transaction: review with id=${req.body.reviewId} does not exist`,
+      });
+      return;
+    }
+  }
+
+  // check if user wallet exists
+  const userWallet = await Wallet.findOne({
+    where: {
+      userId: req.body.userId,
+    },
+  });
+  if (!userWallet) {
+    res.status(404).send({
+      message: `Could not create transaction: wallet with id=${req.body.walletId} does not exist`,
```

Comment:
where is `req.body.walletId` coming from? I thought you use Wallet.findOne to get a wallet belonging to a user by their userId?